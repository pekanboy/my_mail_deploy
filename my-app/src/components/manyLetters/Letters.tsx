import React from 'react';
import {ILetter} from '../../interfaces/ILetter';
import {Letter} from '../Letter/Letter';
import styles from './Letters.module.css';

export interface LettersProps {
    letters: ILetter[]
    selectHandler: (id :number) => void
}

export const Letters: React.FC<LettersProps> = ({letters, selectHandler}) => {
    if (!letters.length) {
        return (
            <div className={styles.zero}>
                У вас нет входящих писем
            </div>
        );
    }

    return (
        <div className={styles.letters}>
            {
                letters?.map((letter) => {
                    return (
                        <Letter
                            key={letter.id.toString()}
                            letter={letter}
                            checkHandler={selectHandler}
                        />
                    );
                },
                )}
        </div>
    );
};
