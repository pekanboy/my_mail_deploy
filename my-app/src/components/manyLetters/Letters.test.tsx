import React from 'react';
import {configure, shallow, ShallowWrapper} from 'enzyme';
import {Letters, LettersProps} from './Letters';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const testData: LettersProps = {
    selectHandler: () => undefined,
    letters: [
        {
            'id': 1,
            'avatar': 'http://i.pravatar.cc/300?u=1',
            'sender': 'Бединге',
            'theme': 'Carolina Hurricanes',
            'snippet': 'Люди всё время меня спрашивают: ' +
                'знаю ли я Тайлера Дёрдена?',
        },
        {
            'id': 2,
            'avatar': 'http://i.pravatar.cc/300?u=2',
            'sender': 'Лиаторп',
            'theme': 'Columbus Blue Jackets',
            'snippet': 'Шесть месяцев я не мог спать. ' +
                'Когда у тебя бессонница — всё нереально; ' +
                'всё очень далеко от тебя, всё это — копия,' +
                ' снятая с копии, которая в свою очередь снята с копии.',
        },
        {
            'id': 3,
            'avatar': 'http://i.pravatar.cc/300?u=3',
            'sender': 'Хемнэс',
            'theme': 'New Jersey Devils',
            'snippet': 'Когда-то мы зачитывались порнографией' +
                ' — теперь каталогами IKEA.',
        }],
};

describe('test Letters block', () => {
    it('letters count no zero', function() {
        const wrapper: ShallowWrapper = shallow(<Letters {...testData} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('letters count zero', function() {
        const data: LettersProps = {...testData, letters: []};
        const wrapper: ShallowWrapper = shallow(<Letters {...data} />);

        expect(wrapper).toMatchSnapshot();
    });
});
