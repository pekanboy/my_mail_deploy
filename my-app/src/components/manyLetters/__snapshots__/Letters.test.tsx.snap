// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Letters block letters count no zero 1`] = `
<div
  className="letters"
>
  <Letter
    checkHandler={[Function]}
    key="1"
    letter={
      Object {
        "avatar": "http://i.pravatar.cc/300?u=1",
        "id": 1,
        "sender": "Бединге",
        "snippet": "Люди всё время меня спрашивают: знаю ли я Тайлера Дёрдена?",
        "theme": "Carolina Hurricanes",
      }
    }
  />
  <Letter
    checkHandler={[Function]}
    key="2"
    letter={
      Object {
        "avatar": "http://i.pravatar.cc/300?u=2",
        "id": 2,
        "sender": "Лиаторп",
        "snippet": "Шесть месяцев я не мог спать. Когда у тебя бессонница — всё нереально; всё очень далеко от тебя, всё это — копия, снятая с копии, которая в свою очередь снята с копии.",
        "theme": "Columbus Blue Jackets",
      }
    }
  />
  <Letter
    checkHandler={[Function]}
    key="3"
    letter={
      Object {
        "avatar": "http://i.pravatar.cc/300?u=3",
        "id": 3,
        "sender": "Хемнэс",
        "snippet": "Когда-то мы зачитывались порнографией — теперь каталогами IKEA.",
        "theme": "New Jersey Devils",
      }
    }
  />
</div>
`;

exports[`test Letters block letters count zero 1`] = `
<div
  className="zero"
>
  У вас нет входящих писем
</div>
`;
